### PR TITLE
ZOB-126 - Fetch data for dashboard cards | Based on #69

### PR DESCRIPTION
**Pull request by oleksandrbalan-etn**
**Do not merge before #69**

- JIRA: https://etnetera.atlassian.net/browse/ZOB-126
- This MR refactors how data are fetched for cards on dashboard

image

### CODE REVIEW

File: lib/services/delivery_service.dart [hunk 1]
```diff
@@ -21,20 +23,69 @@ class DeliveryService {
...
   Future<Delivery?> getDelivery(DateTime date, String donorId) async {
-    final deliveryQuerySnapshot = await _deliveryCollection
-        .where('pickUpFrom', isEqualTo: date)
-        .where('donorId', isEqualTo: donorId)
-        .get();
-
-    if (deliveryQuerySnapshot.docs.isNotEmpty) {
-      return deliveryQuerySnapshot.docs.first.data();
-    }
+    // TODO: Add DeliveryDto to Delivery mapping and return correct instance
```
Comment (fabiawrc) [inline lib/services/delivery_service.dart:26]: Why do we have this TODO? Can we add a link to the JIRA ticket in which this will be addressed?

**Review by fabiawrc — COMMENTED** [review]
Nice changes, I just have one small thing to say...

File: lib/services/delivery_service.dart (hunk 1 above)
Comment (oleksandrbalan-etn) [reply lib/services/delivery_service.dart:26]: Yea, actually there was no JIRA ticket for that one when I have added this TODO 😅 

But it is already addressed in the last PR:

https://github.com/zachran-jidlo/zachranobed/pull/77/files#diff-282ca8dace2f43785f1fc8c96c398e38f56b3b59047e93fa2c3e350961b32c81L33

image

Is it OK for you?

**Review by fabiawrc — APPROVED** [review]

File: lib/services/delivery_service.dart (hunk 1 above)
Comment (fabiawrc) [reply lib/services/delivery_service.dart:26]: Yes. Let's merge it.

**Pull request merged**